### PR TITLE
Improve ownership of gem

### PR DESCRIPTION
**Pull request by rafaelfranca**
This PR improves the gemspec, updates the License, improves the README and test with all versions of Ruby and Rails we support.

It also adds the Rails team as maintainers of this gem.

### CODE REVIEW

File: gemfiles/actionpack-5-2.gemfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+source "https://rubygems.org"
+
+gemspec path: '..'
+
+gem 'rails', '~> 5.1.0'
```
Comment (Edouard-chin) [inline gemfiles/actionpack-5-2.gemfile]:
```suggestion
gem 'rails', '~> 5.2.0'
```

File: cacheable.gemspec [hunk 2]
```diff
@@ -12,21 +12,17 @@ Gem::Specification.new do |s|
   s.summary     = %q{Simple rails request caching}
   s.description = %q{Simple rails request caching}
 
-  s.rubyforge_project = "cacheable"
-
-  s.files         = `git ls-files`.split("\n")
-  s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
-  s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  s.files         = Dir["lib/**/*.rb", "README.rdoc", "LICENSE.txt"]
```
Comment (Morriar) [inline cacheable.gemspec]:
```suggestion
  s.files         = Dir["lib/**/*.rb", "README.md", "LICENSE.txt"]
```

File: Gemfile [hunk 3]
```diff
@@ -2,3 +2,5 @@ source "https://rubygems.org"
 
 # Specify your gem's dependencies in cacheable.gemspec
 gemspec
+
+gem 'rails', '~> 6.0.0'
```
Comment (DerekStride) [inline Gemfile]: I tried to remove the dependency on rails in #27 so you can use it in Rack apps without having to include rails as a dependency.

Comment (rafaelfranca) [reply Gemfile]: This is not a runtime dependency on Rails. It is a development dependency on Rails to make sure it works in Rails applications. It is only used in the test suite.

Comment (rafaelfranca) [reply Gemfile]: Just to be clear I didn't change the dependencies of the gem. I just made explicit in the Gemfile what was already in the gemspec as development dependency so we could chose which version of Rails the gem was tested. We should also have a test suite without Rails so we make sure it works great for rack only apps. I'll work on that.

**Pull request merged**